packages/core/src/odata/common/selectable/collection-field.ts

```diff
@@ -1,23 +1,33 @@
 /* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
 
 import { EntityBase, Constructable } from '../entity';
+import { EdmTypeShared } from '../edm-types';
 import { SelectableEdmTypeField } from './edm-type-field';
 import { Field } from './field';
-import { ComplexTypeField } from './complex-type-field';
-import { SimpleTypeFields } from './simple-type-fields';
+import { ComplexTypeNamespace } from './complex-type-namespace';
 
-export class CollectionField<EntityT extends EntityBase> extends Field<EntityT>
-  implements SelectableEdmTypeField {
+/**
+ * @experimental This is experimental and is subject to change. Use with caution.
+ *
+ * Creates an instance of CollectionField.
+ *
+ * @typeparam EntityT - Type of the entity the field belongs to
+ * @typeparam FieldT - Type of the collection in the field
```

Comment:
Does this mean in Java terms List<String> that FieldT would be String? Then I would change the description to `Type of the entries in the collection` or something like this.